File: android/app/src/main/java/com/smartexpense/android/domain/repository/UserRepository.java
package com.smartexpense.android.domain.repository;

import com.smartexpense.android.data.remote.dto.request.UpdateProfileRequestDto;
import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto;

public interface UserRepository {
    void getMe(ResultCallback<UserProfileResponseDto> callback);
    void updateMe(UpdateProfileRequestDto request, ResultCallback<UserProfileResponseDto> callback);
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/repository/UserRepository.kt
package com.smartexpense.android.domain.repository

import com.smartexpense.android.data.remote.dto.request.UpdateProfileRequestDto
import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto

interface UserRepository {
    suspend fun getMe(): Result<UserProfileResponseDto>
    suspend fun updateMe(request: UpdateProfileRequestDto): Result<UserProfileResponseDto>
}
